sway/input/seat.c

```diff
@@ -751,13 +761,28 @@ static void send_unfocus(struct sway_container *con, void *data) {
 	}
 }
 
+static void save_kbd_layouts(struct sway_view *view) {
+	for (int i = 0; i < view->kbd_layouts->length; ++i) {
+		struct sway_layout_per_kbd *lpk = view->kbd_layouts->items[i];
+		lpk->layout = sway_keyboard_get_layout(lpk->kbd->xkb_state);
+	}
+}
+
 // Unfocus the container and any children (eg. when leaving `focus parent`)
 static void seat_send_unfocus(struct sway_node *node, struct sway_seat *seat) {
 	sway_cursor_constrain(seat->cursor, NULL);
 	wlr_seat_keyboard_clear_focus(seat->wlr_seat);
 	if (node->type == N_WORKSPACE) {
 		workspace_for_each_container(node->sway_workspace, send_unfocus, seat);
 	} else {
+		struct sway_view *view = node->type == N_CONTAINER ?
+			node->sway_container->view : NULL;
+
+		if (view && seat_is_input_allowed(seat, view->surface) &&
+			config->keep_keyboard_layout == KEYBOARD_LAYOUT_PER_WINDOW) {
```

Comment:
Style: two indents for `if` continuation lines, so that we can tell the difference between the condition and the body.

Reply:
done